pkg/relayer/proxy/relay_builders.go

```diff
@@ -10,7 +10,7 @@ import (
 // newRelayRequest builds a RelayRequest from an http.Request.
 func (sync *relayMinerHTTPServer) newRelayRequest(request *http.Request) (*types.RelayRequest, error) {
 	// Replace DefaultMaxBodySize with config options
-	requestBody, err := SafeReadBody(sync.logger, request.Body, defaultMaxBodySize)
+	requestBody, err := SafeRequestReadBody(sync.logger, request, sync.serverConfig.MaxBodySize)
 	if err != nil {
 		return &types.RelayRequest{}, ErrRelayerProxyInternalError.Wrap(err.Error())
```

Comment:
Wrapping the error in `ErrRelayerProxyInternalError` will hide the former to the `Gateway`, we should just return `err`